### PR TITLE
Parser redo

### PR DESCRIPTION
**Pull request by wxxedu**
Reimplement the parser. The detail can be seen in the docs/dev/Parser.md file.

### CODE REVIEW

File: docs/dev/Parsing.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+a list of tokens. Here, since our parsing logic is simple, we do the
+tokenization simply by splitting the input string by white spaces.
+
+We use a `Deque` structure to store the tokens. The benefit of this is that
+we can easily pop the first token from the list, and this would take $O(1)$
+time as compared to $O(n)$ time for a `List`.
+
+The tokenization process is done in the `WingmanParser` class, inside the
+private method `tokenize`.
+
```
Comment (BoAi01) [inline docs/dev/Parsing.md:41]: thanks for the optimization, though it only has a significant effect when the command is really long..

File: docs/dev/Parsing.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Parsing
+
+## Overview
```
Comment (BoAi01) [inline docs/dev/Parsing.md:3]: Thanks, this looks like a great overview and implementation

File: src/main/java/seedu/address/logic/core/WingmanParser.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        if (groupedCommand.isPresent()) {
+            return groupedCommand.get();
+        }
+
+        throw new ParseException("No command group found for operation mode "
+                                     + operationMode);
+    }
+
+    private Optional<Command> parseGroup(OperationMode operationMode,
```
Comment (BoAi01) [inline src/main/java/seedu/address/logic/core/WingmanParser.java:117]: one java doc missing

**Review by BoAi01 — CHANGES_REQUESTED** [review]
The PR looks great! Just add one more javadoc please?

File: docs/dev/Parsing.md (hunk 1 above)
Comment (wxxedu) [reply docs/dev/Parsing.md:41]: really really really really long lmao

File: src/main/java/seedu/address/logic/core/WingmanParser.java (hunk 3 above)
Comment (wxxedu) [reply src/main/java/seedu/address/logic/core/WingmanParser.java:117]: I think private methods do not require java do., currently, it passes the style-check in CI. We can merge first, and latter on I will add the javadoc.

**Pull request merged**